content/number-theory/FastEratosthenes.h

```diff
@@ -0,0 +1,39 @@
+/**
+ * Author: Jakob Kogler, chilli, pajenegod
+ * Date: 2020-04-12
+ * License: CC0
+ * Description: Prime sieve for generating all primes smaller than LIM.
+ * Status: Stress-tested
+ * Time: LIM=1e9 $\approx$ 1.5s
+ * Details: Despite its n log log n complexity, segmented sieve is still faster
+ * than other options, including bitset sieves and linear sieves. This is
+ * primarily due to its low memory usage, which reduces cache misses. This
+ * implementation skips even numbers.
+ *
+ * Benchmark can be found here: https://ideone.com/e7TbX4
+ *
+ * The line `for (int i=idx; i<S+L; idx = (i += p))` is done on purpose for performance reasons.
+ * Se https://github.com/kth-competitive-programming/kactl/pull/166#discussion_r408354338
+ */
+#pragma once
+
+const int LIM = 1e6;
+bitset<LIM> isPrime;
+vi eratosthenes() {
+	const int S = round(sqrt(LIM)), R = LIM / 2;
+	vi pr = {2}, sieve(S+1); pr.reserve(int(LIM/log(LIM)*1.1));
+	vector<array<int, 2>> cp;
```

Comment:
can be shortened to `vector<pii>` without any further changes. alternatively, could use [0]/[1] below to get rid of the C++17 for loop, either way.